lib/bolt/analytics.rb

```diff
@@ -51,15 +53,15 @@ def self.build_client
       NoopClient.new
     end
 
-    def self.config_path(logger)
+    def self.config_path
       path     = File.expand_path(File.join('~', '.puppetlabs', 'etc', 'bolt', 'analytics.yaml'))
       old_path = File.expand_path(File.join('~', '.puppetlabs', 'bolt', 'analytics.yaml'))
 
       if File.exist?(path)
         if File.exist?(old_path)
           message = "Detected analytics configuration files at '#{old_path}' and '#{path}'. Loading "\
                     "analytics configuration from '#{path}'."
-          logger.warn(message)
+          Bolt::Logger.warn_once('duplicate_analytics', message)
```

Comment:
This change was originally because I was re-loading the configfile to write to it, which resulted in double warnings. The final solution doesn't do that, but I think these changes are still more in line with modern Bolt code so I left them.